HPCC-19719 Propagate Custom Binding Params to DESDL Bindings

rpastrana opened the pull request:
- Adds xslt logic to copy any existing Custom Binding Paramas to DESDL Binding

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

rpastrana commented:
@timothyklemm

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: fe9e1558e878da7db86999906538202cc414b4c2
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |88 |88 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |784 |784 |0 |
|test (thor) |715 |715 |0 |
|test (roxie) |828 |828 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   20 sec (00:00:20) |  175 sec (00:02:55) |   59 sec (00:00:59) |    9 sec (00:00:09) |   30 sec (00:00:30) | 1168 sec (00:19:28) |   19 sec (00:00:19) | 1480 sec (00:24:40) |

timothyklemm commented:
Reviewed. Looks good.

richardkchapman commented:
That's not how you spell propagate

rpastrana committed:
HPCC-19719 Propagate Custom Binding Params to DESDL Bindings

- Adds xslt logic to copy any existing Custom Binding Paramas to DESDL Binding

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@richardkchapman please merge

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 5d14e1655fb24350570e2383a3d849c7d1547edc
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |88 |88 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |786 |786 |0 |
|test (thor) |717 |717 |0 |
|test (roxie) |830 |830 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   30 sec (00:00:30) |  170 sec (00:02:50) |   60 sec (00:01:00) |    9 sec (00:00:09) |   29 sec (00:00:29) | 1218 sec (00:20:18) |   18 sec (00:00:18) | 1534 sec (00:25:34) |
